src/GovUk.Education.ExploreEducationStatistics.Content.Services.Tests/DataSetServiceTests.cs

```diff
@@ -0,0 +1,205 @@
+                Assert.Equal(publication.Slug, viewModel.Release.Publication.Slug);
+                Assert.Equal(publication.Topic.Theme.Title, viewModel.Release.Publication.ThemeTitle);
+            }
+        }
+
+        [Fact]
+        public async Task NoRelease()
+        {
+            var publication = _fixture.DefaultPublication()
+                .WithTopic(_fixture.DefaultTopic()
+                    .WithTheme(_fixture.DefaultTheme()))
+                .Generate();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.Publications.AddRangeAsync(publication);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var dataSetService = BuildService(contentDbContext);
+
+                var result = await dataSetService.GetDataSet(
+                    Guid.NewGuid(), Guid.NewGuid());
+
+                result.AssertNotFound();
+
+            }
+        }
+
+        [Fact]
+        public async Task NoFile()
+        {
+            var publication = _fixture.DefaultPublication()
+                .WithReleaseParents(
+                    _fixture.DefaultReleaseParent(publishedVersions: 1,  year: 2000)
+                        .Generate(1))
+                .WithTopic(_fixture.DefaultTopic()
+                    .WithTheme(_fixture.DefaultTheme()))
+                .Generate();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.Publications.AddRangeAsync(publication);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var dataSetService = BuildService(contentDbContext);
+
+                var result = await dataSetService.GetDataSet(
+                    publication.Releases[0].Id, Guid.NewGuid());
+
+                result.AssertNotFound();
+            }
+        }
+
+        [Fact]
+        public async Task ReleaseNotPublished()
+        {
+            var publication = _fixture.DefaultPublication()
+                .WithReleaseParents(
+                    _fixture.DefaultReleaseParent(publishedVersions: 0, draftVersion: true, year: 2000)
+                        .Generate(1))
+                .WithTopic(_fixture.DefaultTopic()
+                    .WithTheme(_fixture.DefaultTheme()))
+                .Generate();
+
+            var releaseFile = _fixture.DefaultReleaseFile()
+                .WithRelease(publication.Releases[0])
+                .WithFile(_fixture.DefaultFile())
+                .Generate();
```

Comment:
Fyi if you want to you can avoid two `.Generate()` lines here and elsewhere in this file where you use `.Generate()` by using the implicit operator already defined by `Generator`:

```csharp
    /// <summary>
    /// Automatically generate an instance of T from the generator
    /// using implicit type conversion.
    /// </summary>
    public static implicit operator T(Generator<T> generator)
    {
        return generator.Generate();
    }
```

As an example:

```csharp
Publication publication = _fixture.DefaultPublication()
      .WithReleaseParents(
          _fixture.DefaultReleaseParent(publishedVersions: 0, draftVersion: true, year: 2000)
              .Generate(1))
      .WithTopic(_fixture.DefaultTopic()
          .WithTheme(_fixture.DefaultTheme()));
```